### ISSUE TITLE
Accelerometer update makes UI laggy

### ISSUE THREAD

**Issue by chocke** [bpisano/Sticker#3]
Unfortunately when i use it with accelerometer on iPhone, it makes the animation lagging. I will take a look when I have some spare time.

**Comment by bpisano** [comment]
Thank you for reporting. Do you have any sample code I could use to reproduce the issue? Also, what iOS version do you target?

**Comment by chocke** [comment]
> Thank you for reporting. Do you have any sample code I could use to reproduce the issue? Also, what iOS version do you target?

Xcode - 16.1 (16B40)
Deployment target - iOS 18.1
Device - iPhone 14 Pro iOS 18.1

`struct ContentView: View {
    var body: some View {
        VStack {
            Image(.apple)
                .resizable()
                .frame(width: 200, height: 200)
                .stickerEffect()
                .stickerMotionEffect(.accelerometer)
        }
        .padding()
    }
}`

![apple](https://github.com/user-attachments/assets/695eac23-676e-4c04-b1a9-383aff826ebf)

**Comment by bpisano** [comment]
The _lagginess_ you described was due to the CoreMotion accelerometer `updateInterval` property set to high. I released version 1.3.0 that lower this value to get a nice effect by default.

But decreasing the `updateInterval` can also have battery performance on the device. That's why you should consider increasing this value, but instead wrap your component inside an animation to get a smooth movement.

```swift
struct ContentView: View {
    var body: some View {
        Image(systemName: "applelogo")
            .resizable()
            .aspectRatio(contentMode: .fit)
            .frame(width: 200, height: 200)
            .animation(.smooth) { view in
                view
                    .stickerEffect()
                    .stickerMotionEffect(.accelerometer(updateInterval: 0.1))
            }
    }
}
```
